cherab/tools/observers/bolometry.py

```diff
@@ -687,6 +695,350 @@ def etendue_single_run(_):
+    :ivar array pixels_as_foils: A 2D array of pixels as individual BolometerFoil objects,
+      useful for calling BolometerFoil methods on each pixel (e.g. sightline tracing).
+      The array is indexed by (column, row).
+    :ivar Vector3D sightline_vectors: A 2D array of vectors that point from the centre
+      of each pixel on the foil detector to the centre of the slit. Defines the
+      effective sightline vectors of the pixels of the detector. The list is indexed by
+      (column, row).
+
+    .. code-block:: pycon
+
+       >>> from raysect.core import Point3D, Vector3D
+       >>> from raysect.optical import World
+       >>> from cherab.tools.observers import BolometerIRVB
+       >>>
+       >>> world = World()
+       >>>
+       >>> # construct basis vectors
+       >>> basis_x = Vector3D(1, 0, 0)
+       >>> basis_y = Vector3D(0, 1, 0)
+       >>> basis_z = Vector3D(0, 0, 1)
+       >>>
+       >>> # specify a detector, you need already created slit and camera objects
+       >>> width = 0.0025
+       >>> pixels = (10, 20)
+       >>> centre_point = Point3D(0, 0, -0.08)
+       >>> detector = BolometerIRVB("ch#1", centre_point, basis_x, basis_y, width, pixels, slit, parent=camera)
+    """
+
+    def __init__(self, detector_id, centre_point, basis_x, basis_y, width, pixels, slit,
+                 parent=None, units="Power", accumulate=False, curvature_radius=0):
+
+        # perform validation of input parameters
+
+        if not isinstance(width, (float, int)):
+            raise TypeError("width argument for BolometerIRVB must be of type float/int.")
+        if width < 0:
+            raise ValueError("width argument for BolometerIRVB must be greater than zero.")
+
+        if not isinstance(slit, BolometerSlit):
+            raise TypeError("slit argument for BolometerIRVB must be of type BolometerSlit.")
+
+        if not isinstance(centre_point, Point3D):
+            raise TypeError("centre_point argument for BolometerIRVB must be of type Point3D.")
+
+        if not isinstance(curvature_radius, (float, int)):
+            raise TypeError("curvature_radius argument for BolometerIRVB "
+                            "must be of type float/int.")
+        if curvature_radius < 0:
+            raise ValueError("curvature_radius argument for BolometerIRVB "
+                             "must not be negative.")
+
+        if not isinstance(basis_x, Vector3D):
+            raise TypeError("The basis vectors of BolometerIRVB must be of type Vector3D.")
+        if not isinstance(basis_y, Vector3D):
+            raise TypeError("The basis vectors of BolometerIRVB must be of type Vector3D.")
+
+        basis_x = basis_x.normalise()
+        basis_y = basis_y.normalise()
+        normal_vec = basis_x.cross(basis_y)
+        self._slit = slit
+        self._curvature_radius = curvature_radius
+        self._accumulate = accumulate
+
+        # setup root bolometer foil transform
```

Comment:
This breaks the raysect and cherab  convention of supplying an AffineMatrix to define the space relative to the parent. Please don't do this as it forces a specific representation on the user.

Reply:
This is now done: the IRVB constructor takes an affine transform instead of centre point and basis vectors. The docstring gives an example of how to produce this transform for a given centre and basis, for those used to the BolometerFoil way of doing things.